src/main/java/com/basho/riak/client/core/query/timeseries/TableDefinition.java

```diff
@@ -0,0 +1,163 @@
+package com.basho.riak.client.core.query.timeseries;
+
+import java.util.*;
+
+/**
+ * Holds a definition for a Table in Time Series Riak.
+ * Immutable once created.
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.4
+ */
+public class TableDefinition
+{
+    private final String tableName;
+    private final LinkedHashMap<String, FullColumnDescription> fullColumnDescriptions = new LinkedHashMap<>();
+    private final ArrayList<FullColumnDescription> partitionKeys = new ArrayList<>();
```

Comment:
I lean the other way, we can easily create it while in the constructor, and then we have an entire immutable object once we're done.  If we do it later then we're saving 8? object references, and introducing more code and possible race conditions later.